src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -340,13 +340,17 @@ private ClassName resolve(String simpleName) {
 
   /** Returns the class named {@code simpleName} when nested in the class at {@code stackDepth}. */
   private ClassName stackClassName(int stackDepth, String simpleName) {
-    ClassName className = ClassName.get(packageName, typeSpecStack.get(0).name);
+    ClassName className = ClassName.get(getPackageNameNullSafe(), typeSpecStack.get(0).name);
     for (int i = 1; i <= stackDepth; i++) {
       className = className.nestedClass(typeSpecStack.get(i).name);
     }
     return className.nestedClass(simpleName);
   }
 
+  private String getPackageNameNullSafe() {
```

Comment:
This prevents the NPE that is thrown, when building nested `TypeSpec`s and the constructor of `TypeSpec` contains a `toString()` call as its last statement.

These 3 tests will fai in `TypeSpecTest.java`:
* `typeVariable()`
* `nestedClass()`
* `referencedAndDeclaredSimpleNamesConflict()`